### ISSUE TITLE
No module named '_dci_cuda'

### ISSUE THREAD

**Issue by zapaishchykova** [niopeng/HyperRIM#1]
Hello! Thanks for releasing the code! I got the following error, do you know what could be a possible issue?
```
python train.py -opt options/train/[train_sr.json, train_decompression.json]

Traceback (most recent call last):
  File "train.py", line 17, in <module>
    from sampler import generate_code_samples
  File "D:\HyperRIM\code\sampler.py", line 4, in <module>
    from dciknn_cuda.dciknn_cuda import DCI
  File "D:\HyperRIM\code\dciknn_cuda\dciknn_cuda\__init__.py", line 18, in <module>
    from .core import DCI
  File "D:\HyperRIM\code\dciknn_cuda\dciknn_cuda\core.py", line 19, in <module>
    from _dci_cuda import _dci_new, _dci_add, _dci_query, _dci_clear, _dci_reset, _dci_free
ModuleNotFoundError: No module named '_dci_cuda'
```

**Comment by niopeng** [comment]
Hi! Did you clone the dciknn_cuda submodule and run 'python* setup.py install' in that folder? 
*Note that the python here should match the one that is used to run 'train.py'. i.e. if your training command is 'python3 train.py xx', then the setup should be done with 'python3 setup.py install'.

**Comment by zapaishchykova** [comment]
did you build it on ubuntu? i cannot build it on windows 10 :(
```
fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory
error: command 'C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v10.2\\bin\\nvcc.exe' failed with exit status 2
```

**Comment by niopeng** [comment]
Could you try commenting out '#include unistd.h' in 'dciknn_cuda/src/util_kernel.cu' and 'dciknn_cuda/include/util.h'? Would it compile then?

**Comment by zapaishchykova** [comment]
I commented, now i have:
```
Creating library build\temp.win-amd64-3.6\Release\./src\_dci_cuda.cp36-win_amd64.lib and object build\temp.win-amd64-3.6\Release\./src\_dci_cuda.cp36-win_amd64.exp
util_kernel.obj : error LNK2001: unresolved external symbol cublasCreate_v2
util_kernel.obj : error LNK2001: unresolved external symbol cublasDestroy_v2
util_kernel.obj : error LNK2001: unresolved external symbol cublasSgemm_v2
build\lib.win-amd64-3.6\_dci_cuda.cp36-win_amd64.pyd : fatal error LNK1120: 3 unresolved externals
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\link.exe' failed with exit status 1120
```

tried a solution from here https://stackoverflow.com/questions/50494411/error-command-c-visual-studio-2017-community-vc-tools-msvc-14-14-2642 to install a c++ connector, still same issue

**Comment by niopeng** [comment]
I think it might not be an issue with the linker, but with the culas library. Could you try some solutions listed here:
1. https://forums.developer.nvidia.com/t/trouble-using-cublas-new-to-cuda-unresolved-external-symbols/7714
2. https://stackoverflow.com/questions/34032878/cublas-error-link-2019?noredirect=1&lq=1
3. https://stackoverflow.com/questions/2061715/unresolved-external-symbols-in-beginners-cuda-program
4. https://forums.developer.nvidia.com/t/error-lnk2019-unresolved-external-symbol/11366/14

Basically, I suspect that the cublas library is not linked properly.

**Comment by AlexiaJM** [comment]
I have the same issue :(

**Comment by AlexiaJM** [comment]
You can see the error in Google Colab (make sure to set to GPU) with the following commands:
> !git clone https://github.com/niopeng/dciknn_cuda.git
> cd dciknn_cuda
> !python setup.py install
> import dciknn_cuda # Error here

See https://colab.research.google.com/drive/1DZAwiWfBs2_VuIjSMcL_zYgcKx3ej7Ie?usp=sharing

**Comment by niopeng** [comment]
Hi Alexia. I think the problem that you raised is different. Could you verify that after running `!python setup.py install` try `!python example.py` and whether that works? I see in this [post](https://github.com/openphilanthropy/unrestricted-adversarial-examples/issues/42) that you would need to restart your kernel and you can import the module by `from dciknn_cuda.dciknn_cuda import DCI`

**Comment by AlexiaJM** [comment]
😹I feel silly now, restarting the kernel was all it took. Thanks!
